daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/IdString.scala

```diff
@@ -111,6 +111,19 @@ object IdString {
   import Ref.{Name, Party}
   implicit def `Name order instance`: Order[Name] = Order fromScalaOrdering Name.ordering
   implicit def `Party order instance`: Order[Party] = Order fromScalaOrdering Party.ordering
+
+  private[data] def asciiCharsToRejectionArray(s: Iterable[Char]): Array[Boolean] = {
+    val array = Array.fill(0x80)(true)
```

Comment:
Do you think it's worth investigating an alternative implementation with two 64-bit integers and C-style bit fiddling instead of an array of 128 booleans?

Reply:
I do not think it is worth investigating. It will be more complex, probably slower and will save us less than 1KB of memory.

Reply:
I wasn't looking for memory improvements but rather for speedups. But if you don't anticipate great gains but plenty of complexity, let's not look into that now. If we ever get told again that checking names shows up big in profiling data, we can still have another look.

Reply:
I don't see how you can make it faster with but manipulation. I am a bit curious about what I you have in mind. 
Don't think it is worthwhile to send time to satisfy my curiosity.

Reply:
What we need here is basically a set of ASCII characters. The implementation in this PR uses an array of 128 bytes for this. My idea would be to use two 64-bit integers instead. My hope is that both of them will reside in registers during execution and we don't need to consult the memory at all to find out if a character is valid. I also hope that the compiler would inline all the bit fiddling.

Unfortunately, I'm struggling to express my idea in Scala, so let me throw some Rust at you for the sake of illustration:
```rust
struct AsciiCharSet {
    lower: u64,
    upper: u64,
}

impl AsciiCharSet {
    fn new() -> Self {
        Self { lower: 0, upper: 0 }
    }
    
    fn insert(&mut self, c: char) {
        let c = u32::from(c);
        assert!(c < 0x80);
        if c < 0x40 {
            self.lower |= 1 << c;
        } else {
            self.upper |= 1 << (c - 0x40);
        }
    }
    
    fn contains(&self, c: char) -> bool {
        let c = u32::from(c);
        assert!(c < 0x80);
        if c < 0x40 {
            self.lower & (1 << c) != 0
        } else {
            self.upper & (1 << (c - 0x40)) != 0
        }
    }
}
```

Does that make sense?

Reply:
Thank you for elaborating, I understand you better now. You expect avoiding access to the memory will overcome the complexity of bit manipulation. 

Indeed I do not think it is worthwhile to investigate more. I think it will be very difficult to measure any difference of speed (specially in the JVM). 

As a matter of comparison a lot of UTF8 validators/converter, I am aware of, use a lookup table at some point (see for instance standard [go](https://github.com/golang/go/blob/master/src/unicode/utf8/utf8.go) or [rust](https://doc.rust-lang.org/src/core/str/validations.rs.html) validators or [C++ protobuf library](https://github.com/protocolbuffers/protobuf/blob/c0b79c5a398f4eec51ef92784e9d940febdccc52/src/google/protobuf/stubs/structurally_valid.cc).

Interestingly, neither [the standard Java UTF8 translator](https://github.com/openjdk/jdk15/) nor [Java protobuf library UTF8 validator](https://github.com/protocolbuffers/protobuf/blob/46a48e49aa8357bbeee8040819a35e59880e329a/java/core/src/main/java/com/google/protobuf/Utf8.java) used lookup table.

Reply:
I still think the bit fiddling is faster because it can all be done in registers. Accessing the array needs to first compute a memory address, which is very likely itself already as expensive as the bit fiddling, and then retrieve data from memory, which is surprisingly slow.

But I don't think it will have a measurable impact on the system as a whole in the current state of affairs. I don't know which percentage of time of a request we spend validating identifiers with the current implementation but I doubt it's significant. Hence, no improvement to this validation will have a significant overall impact anyway. I don't believe we need to do anything here right now, I was just curious. 😃